src/lib.rs

```diff
@@ -10,21 +11,66 @@
 /// ```
 ///
 /// #### After:
-/// ```    
+/// ```ignore
 /// let result: Result<T, E> = try_block! {
 ///    let a = do_one(x)?;
 ///    let b = do_two(a)?;
-///    Ok(b)
+///    b
 /// };
 /// ```
 
 #[macro_export]
 macro_rules! try_block {
     { $($token:tt)* } => {{
-        let l = || {
-            $($token)*
-        };
-        l()
+        ( || $crate::FromOk::from_ok(
+            { $($token)* }
+        ))()
     }}
 }
 
+/// A type that can Ok-wrap some value, for use in the [`try_block`] macro.
+pub trait FromOk {
```

Comment:
You can actually avoid the need for this trait with a bit of trickery.  `Try::from_output` is unstable, but you can get to it via `try_fold`: <https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0b07b516d97996ac386886ee21d8c561>
```rust
macro_rules! do_ok_wrapping {
    ($e:expr) => {
        std::iter::empty()
            .try_fold($e, |_, ()| unreachable!())
    };
}

fn main() {
    let _: Option<i32> = do_ok_wrapping!(3);
    let _: std::io::Result<i32> = do_ok_wrapping!(5);
}
```

(That might make the error messages horrible, though.)

Reply:
You can even remove the panic by using the never type instead of unit.
```rust
std::iter::empty()
    .try_fold($e, |_, x: !| match x { })
```

I think this repo is unmaintained so I'm going to fork it, and I'll definitely use your idea.

Reply:
Oh, nice touch!  An uninhabited enum -- no need for `!` specifically -- is a great idea.